pandas/tests/io/test_excel.py

```diff
@@ -544,6 +544,18 @@ def test_date_conversion_overflow(self):
         result = self.get_exceldf('testdateoverflow')
         tm.assert_frame_equal(result, expected)
 
+    # GH10559: Minor improvement: Change to_excel "sheet_name" to "sheetname"
+    # GH10969: DOC: Consistent variable names (sheetname vs sheet_name, issue 10559)
+    # GH12604: CLN GH10559 Rename sheetname variable to sheet_name for consistency
+    def test_sheet_name_and_sheetname(self):
+        dfref = self.get_csv_refdf('test1')
+        df1 = self.get_exceldf('test1', sheet_name='Sheet1')    # doc
+        df2 = self.get_exceldf('test1', sheetname='Sheet2')     # bkwrds compat
```

Comment:
You can assert that a warning with the `tm.assert_produces_warning` context manager.